### ISSUE TITLE
Problems loading Corda Node on an Android App

### ISSUE THREAD

**Issue by yshprasanth** [corda/corda#1730]
I am working on a POC to deploy a Corda Node on an Android App.

First step for is to ensure the Corda libraries are included in our APK and deployed to Android Emulator.

In this process, I have add below dependency in my app's gradle.build file:

	implementation("net.corda:corda-node:0.14.0", {
        exclude group: 'com.github.bft-smart', module: 'library'
        exclude group: 'com.github.corda.crash', module: 'crash.shell'
        exclude group: 'org.jetbrains.exposed', module: 'exposed'
        //exclude group: 'io.netty'
        exclude group: 'io.netty', module: 'netty-transport'
        exclude group: 'io.netty', module: 'netty-handler'
        transitive = true
    })
	
	packagingOptions {
        // Excluding the below files to ensure the build of APK with Corda Node library works ok
        exclude 'META-INF/INDEX.LIST'
        exclude 'META-INF/DEPENDENCIES'
        exclude 'META-INF/org/apache/logging/log4j/core/config/plugins/Log4j2Plugins.dat'
        exclude 'META-INF/io.netty.versions.properties'
    }
	
The APK build was fine after the above configuration with multiple dex files. However, when I am starting the APK, I see the below error:

	_java.io.IOException: Failed to open dex files from /data/app/com.example4.corda.cordamobileapp4-Ma2gE0QLACHd94ICti9wGg==/split_lib_dependencies_apk.apk because: Failure to verify dex file '/data/app/com.example4.corda.cordamobileapp4-Ma2gE0QLACHd94ICti9wGg==/split_lib_dependencies_apk.apk': _Invalid method name: 'owned by'_
                 
This is because of having the method names as `issued by` an `owned by` with back ticks and spaces in below files:

		○ corda\finance\src\main\kotlin\net\corda\finance\contracts\asset\Cash.kt
		○ corda\finance\src\main\kotlin\net\corda\finance\contracts\asset\Obligation.kt
		○ corda\finance\src\main\kotlin\net\corda\finance\contracts\CommercialPaper.kt
		○ corda\finance\src\main\kotlin\net\corda\finance\Currencies.kt
		○ corda\samples\irs-demo\src\test\kotlin\net\corda\irs\api\NodeInterestRatesTest.kt

		
I have made the changes in my local copy, to remove space, re-compile to JAR. 

I was then able to launch the APP with the modified corda-finances.jar.

Can you please have a look and modify these method names in the next release or provide an alternate solution, if possible.

**Comment by mikehearn** [comment]
Yeah, we can do that. These method name are not in API stabilised modules.

That said, the Corda node is not intended to run on Android. It will perform *VERY SLOWLY*. Can you tell us more about what you want to do? Corda for phones is something I've thought about a bit, but just bringing up a full blown node on a phone is the wrong way to go.

**Comment by miguelangelvaldivia** [comment]
Hi Mike.  I am working with Sri on this. We appreciate that the solution is not built for Android but would like to test the idea of moving from a client-server architecture into a serverless one.  We picked up Corda as we figured the API interfacing would come handy when dealing with our legacies.  In order to do that we need to turn our clients (phones) into servers.  Easier said that done.  But your thoughts on this would be very welcome.
